content/realtime/channels.textile

```diff
@@ -813,6 +814,55 @@ Previously registered listeners can be removed individually or all together.
     stateChangeListener.cancel();
 ```
 
+
+h3(#channel-detach). Detaching from a channel
+
+A client can detach from a channel so that they no longer receive any messages published to that channel. Detaching is different to unsubscribing from a channel as "@unsubscribe()@":#unsubscribe is a client-side operation, meaning Ably is unaware of whether or not a client has unsubscribed from the channel. Messages will continue to be streamed to the client until "@detach()@":#detach is called.
+
+bc[jsall]. channel.detach();
+channel.on('detached', function(stateChange) {
+  console.log('detached from the channel ' + channel.name);
+};
+
+bc[ruby]. channel.detach
+channel.on(:detached) do |channel_state_change|
+  puts "detached from the channel #{channel.name}"
+end
+
+bc[java]. channel.detach();
+channel.on(ChannelEvent.detached, new ChannelStateListener() {
+  @Override
+  public void onChannelStateChanged(ChannelState state, ErrorInfo reason) {
+    System.out.println("detached from the channel " + channel.name);
+    if (reason != null) System.out.println(reason.toString());
+  }
+});
+
+bc[csharp]. Channel.Detach();
+channel.On(ChannelEvent.Detached, stateChange => {
+  Console.WriteLine("detached from the channel " + channel.Name)
+});
+
+bc[objc]. [channel detach]
+[channel on:ARTChannelEventDetached callback:^(ARTChannelStateChange *stateChange) {
+  NSLog(@"detached from the channel ", channel.name);
+}];
+
+bc[swift]. channel.detach()
+channel.on(.detached) { stateChange in
+  print("detached from the channel \(channel.name)")
+}
+
+bc[flutter]. channel.detach();
+final stateChangeListener = channel
+  .on(ably.ChannelEvent.detached)
+  .listen((ably.ChannelStateChange state) {
+    print('detached from the channel ${channel.name}');
+  }
+);
+
+A channel will automatically close when there are no more realtime clients attached to it and approximately one minute has passed since the last client detached and since the last message was published to the channel.
```

Comment:
Is this correct? I'm thinking that, for a message to be published, a client must be attached to the channel to do the publishing. Therefore, the last client leaving will always be later than the last message being published? 🤔

Reply:
This is true as far as I'm aware, yes. You can always publish by REST so this can occur after you no longer have any realtime attachments. 

Found the [FAQ](https://faqs.ably.com/how-does-ably-count-peak-channels) I read this in for reference.

Reply:
Correct, REST publishes can take place without attachment. (And also equivalently, realtime publishes via a transient attach - technically they are attaches under the hood, but clients don't see that - they just call `publish()`).

Reply:
Thanks for the confirmation, Paddy.